modules/core/src/main/java/org/apache/ignite/internal/processors/cache/CacheGroupContext.java

```diff
@@ -1191,7 +1194,30 @@ public void onReconnected() {
      * WAL enabled flag.
      */
     public boolean walEnabled() {
-        return localWalEnabled && globalWalEnabled;
+        return walEnabled(false);
+    }
+
+    /**
+     * @param forCdc {@code True} record can be consumed by {@link CdcConsumer}.
+     * @return {@code True} if {@link DataRecord} should be logged into WAL.
```

Comment:
It also returns true for every other records if WAL is enabled.

Reply:
Thanks. Fixed.